client/src/Components/PostPage/PostPage.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import NavbarComponent from '../Navbar/Navbar.js';
+
+const PostPage = ({ match }) => {
+
+	const [isLoading, setIsLoading] = useState(true);
+	const [data, setData] = useState();
+
+	useEffect(() => {
+		fetch(`/posts/${match.params.id}`, {}).then((res) => 
+			res.json()
+		).then((res) => {
+			if (typeof res.title === 'string' && typeof res.body == 'string'){
+				setData(res);
+				setIsLoading(false);
+			}
+		}).catch((error) => 
+			console.log(error)
+		);
+	}, [match.params.id]);
```

Comment:
By the way, you could turn these calls into custom hooks that would just return variables. And all this code move to the hook.
```suggestion
	const { data, loading } = useSinglePostFetch(id);
```